### PR TITLE
Updated file_io.rst to give a MVP for production

### PR DESCRIPTION
**Pull request by jfkcooper**
This PR adds alot more details to the `fileio_usage.rst` to provide better information on how one would use the classes of `orsopy` to create a bespoke file writer for an instrument.
The code (when ran as a whole) will create a working `.ort` file with I *think* all necessary entries.

### CODE REVIEW

File: docs/fileio_usage.rst [hunk 1]
```diff
@@ -3,60 +3,104 @@ Usage
...
+It is generally a good idea to populate the components as particular steps occur in the reduction process.
+In this example we fill out the experiment details and then wrap them in the appropriate class.
 
-.. code-block:: python 
+.. code-block:: python
 
-    header.data_source.experiment.probe = 'neutron'
+    title = 'my title'
+    instrument = 'PLATYPUS'
+    start_date = datetime.date.today()  # This needs to be a datetime object
```
Comment (arm61) [inline docs/fileio_usage.rst:26]:
```suggestion
    start_date = datetime.date.today()  # This needs to be an object of the type datetime
```

File: docs/fileio_usage.rst [hunk 2]
```diff
@@ -3,60 +3,104 @@ Usage
...
+    probe = 'neutron'
+    facility = 'ESS'
+    proposalID = 999999
+    doi = f'10.5286/ISIS.E.RB{99999}'
```
Comment (arm61) [inline docs/fileio_usage.rst:30]: Have you deliberately mixed ISIS, ESS and ANSTO?

File: docs/fileio_usage.rst [hunk 3]
```diff
@@ -3,60 +3,104 @@ Usage
...
+
+    experiment = Experiment(title, instrument, start_date, probe, facility, proposalID, doi)
 
 Full details of the different components that can be populated can be found in the `documentation`_ here or in the `file format specification`_.
 Note that this specification includes information regarding the **required** and optional components to be included for a file to be considered a **valid** .ort file.
-It is not possible to write a .ort file without defining the columns present in the dataset, in this example we will have four columns of data, namely q, R, dR and dq (the final column is a description of the resolution function). 
-Columns are defined as follows, using the :code:`orsopy.fileio.base.Column` and :code:`orsopy.fileio.base.ErrorColumn` class objects (note that there are other `base classes`_ that can be used for a variety of objects).
+What we show here is not the minimum needed to write an `.ort file, rather the minium to fill out all the major fields with information which is likely to be available at time of writing.
```
Comment (arm61) [inline docs/fileio_usage.rst:36]:
```suggestion
What we show here is not the minimum needed to write an `.ort` file, rather the minium to fill out all the major fields with information which is likely to be available at time of writing.
```

File: docs/fileio_usage.rst [hunk 4]
```diff
@@ -3,60 +3,104 @@ Usage
...
+    files = ['test_data.raw']  # A list of all raw files used reduced
+
+    measurment = Measurement(instrument_settings, files)
+
+We can now wrap together the user, experiment, sample and measurement information into a single object.
+
+.. code-block:: python
+    data_source_info = DataSource(user, experiment, sample, measurment)
+
+It is not possible to write a .ort file without defining the columns present in the dataset, in this example we will have four columns of data, namely q, R, dR and dq (the final column is a description of the resolution function).
```
Comment (arm61) [inline docs/fileio_usage.rst:68]:
```suggestion
It is not possible to write an `.ort` file without defining the columns present in the dataset, in this example we will have four columns of data, namely q, R, dR and dq (the final column is a description of the resolution function).
```

File: docs/fileio_usage.rst [hunk 5]
```diff
@@ -3,60 +3,104 @@ Usage
...
+Columns are defined as follows, using the :code:`orsopy.fileio.base.Column` and :code:`orsopy.fileio.base.ErrorColumn` class objects.
+
+.. code-block:: python
 
-.. code-block:: python 
-    
     q_column = fileio.base.Column(name='Qz', unit='1/angstrom', physical_quantity='wavevector transfer')
+    # Note that the unit is not fixed to 1/angstrom by the standard, but currently the data loader interprets all values as 1/angstrom irrespective of what is stated here
```
Comment (arm61) [inline docs/fileio_usage.rst:74]: Is this true?

**Review by arm61 — APPROVED** [review]

**Comment by arm61** [comment]
Some small comments (and one important missing `

File: docs/fileio_usage.rst (hunk 2 above)
Comment (jfkcooper) [reply docs/fileio_usage.rst:30]: yes

File: docs/fileio_usage.rst (hunk 5 above)
Comment (jfkcooper) [reply docs/fileio_usage.rst:74]: sadly yes. there is no unit processing that I could find in the data loader, and all analysis packages currently read data specified with 1/nanometer as 1/angstrom

Comment (jfkcooper) [reply docs/fileio_usage.rst:74]: oh, but no!
I was just not using the right unit!!!

Comment (jfkcooper) [reply docs/fileio_usage.rst:74]: `1/nm` not `1/nanometer`

**Comment by jfkcooper** [comment]
Many thanks @arm61!

**Pull request merged**